rotkehlchen/exchanges/kraken.py

```diff
@@ -1162,11 +1088,22 @@ def history_event_from_kraken(
                         event_subtype=event_subtype,
                     )
                     if event.event_type in (HistoryEventType.RECEIVE, HistoryEventType.SPEND):
-                        receive_spend_events[event.event_identifier].append((idx, event))
+                        receive_spend_events[event.event_identifier].append((idx, event))  # type: ignore  # will only be HistoryEvent for receive/spend
                     else:
                         group_events.append((idx, event))
                 if event_type != HistoryEventType.INFORMATIONAL and fee_amount != ZERO:  # avoid processing ignored events with fees that were converted to informational  # noqa: E501
-                    group_events.append((idx, HistoryEvent(
+                    group_events.append((idx, AssetMovement(
+                        timestamp=timestamp,
+                        location=Location.KRAKEN,
+                        event_type=event_type,  # type: ignore  # will be deposit or withdrawal
+                        asset=asset,
+                        balance=Balance(abs(fee_amount)),
+                        unique_id=identifier,
+                        is_fee=True,
+                    ) if event_type in {
+                        HistoryEventType.DEPOSIT,
```

Comment:
you are doing same thing as above here. I think this function is getting too complicated and needs to be rethought. Lots of code is repeating

Reply:
Also I believe you are not converting them properly. In line 1045 we have

```python
                elif event_type == HistoryEventType.DEPOSIT:
                    event_subtype = HistoryEventSubType.DEPOSIT_ASSET
                elif event_type == HistoryEventType.WITHDRAWAL:
                    event_subtype = HistoryEventSubType.REMOVE_ASSET
``` 

I believe you should catch those two and convert them to assetMovements instead of history events